### ISSUE TITLE
Investigate none_mtls_both vs. none_plaintext_both

### ISSUE THREAD

**Issue by carolynhu** [istio/istio#22416]
Currently, none_plaintext_both has larger/obvious latency than none_mtls_both, which should be reversed. We need to investigate the reason and fix it.

**Comment by incfly** [comment]
http://perf.dashboard.istio.io/benchmarks/latency_vs_connection/

hmm, seems like the issue existed since very early of March, 0306 around.

**Comment by incfly** [comment]
I did a local run in my cluster, plaintext is better than mtls. Here is [gist](https://gist.github.com/incfly/2fb2988867e072d3c4b060b5e5572236), data.

e.g. 

## 8 connection

mtls
```
# target 50% 0.00272268
# target 75% 0.0029291
# target 90% 0.00320823
# target 99% 0.00350828
# target 99.9% 0.00405612
```

plaintext
```
# target 50% 0.00249595
# target 75% 0.00276632
# target 90% 0.00292924
# target 99% 0.00337642
# target 99.9% 0.0037672
```

## 16 connection

```
mtls
# target 50% 0.00474316
# target 75% 0.00524581
# target 90% 0.00573406
# target 99% 0.00646893
# target 99.9% 0.0069774
Sockets used: 16 (for perfect keepalive, would be 16)

plaintext
# target 50% 0.00421123
# target 75% 0.00457216
# target 90% 0.004904
# target 99% 0.00583192
# target 99.9% 0.00599256
Sockets used: 16 (for perfect keepalive, would be 16)
```

**Comment by incfly** [comment]
This is how I setup the test.

```
export NAMESPACE=twopods-istio
export INTERCEPTION_MODE=REDIRECT
export ISTIO_INJECT=true
cd ../benchmark
./setup_test.sh
```

```bash
bash ./configs/istio/plaintext/prerun.sh
python runner/runner.py --config_file ./configs/istio/plaintext/latency.yaml 2>&1 | tee plaintext.log
bash ./configs/istio/plaintext/postrun.sh
python runner/runner.py --config_file ./configs/istio/plaintext/latency.yaml 2>&1 | tee mtls.log

**Comment by incfly** [comment]
so roughly the test setup itself seems correct. would it be possible the `run_benchmark_jobs.sh` has some mis-configuration? or the I ran this manually myself. are there logs of `run_benchmark_jobs.sh` we can look?

**Comment by carolynhu** [comment]
@incfly here is a log you can reference to: https://storage.googleapis.com/istio-prow/logs/daily-performance-benchmark/193/build-log.txt

**Comment by incfly** [comment]
Hmm from the log above, you can see the data is consistent. at same 32 socket case, none(both), which consistently has shorter latency compared with mtls.

@carolynhu might want to check how the plotting graph is calculated?

```
# target 50% 0.00519141
# target 75% 0.00567687
# target 90% 0.00596814
# target 99% 0.00690593
# target 99.9% 0.00763796
Sockets used: 32 (for perfect keepalive, would be 32)
Jitter: false
Code 200 : 240000 (100.0 %)
Response Header Sizes : count 240000 avg 167.00023 +/- 0.01514 min 167 max 168 sum 40080055
Response Body/Total Sizes : count 240000 avg 1191.0002 +/- 0.01514 min 1191 max 1192 sum 285840055
All done 240000 calls (plus 32 warmup) 4.947 ms avg, 1000.0 qps
Successfully wrote 3650 bytes of Json data to 2020-03-23-030057_1d9e87ff_qps_1000_c_32_1024_none_both.json


# target 50% 0.00742435
# target 75% 0.00856629
# target 90% 0.0101297
# target 99% 0.012276
# target 99.9% 0.0124907
Sockets used: 32 (for perfect keepalive, would be 32)
Jitter: false
Code 200 : 240000 (100.0 %)
Response Header Sizes : count 240000 avg 167.00075 +/- 0.0273 min 167 max 168 sum 40080179
Response Body/Total Sizes : count 240000 avg 1191.0007 +/- 0.0273 min 1191 max 1192 sum 285840179
All done 240000 calls (plus 32 warmup) 7.295 ms avg, 1000.0 qps
Successfully wrote 3936 bytes of Json data to 2020-03-23-035817_184a6c9c_qps_1000_c_32_1024_none_plaintext_both.json
```

**Comment by carolynhu** [comment]
@incfly if so, we need to figure out the problem with our setup of either `none_mtls_both` or `none_plaintext_both` or `both of them`.

**Comment by carolynhu** [comment]
cc @richardwxn

**Comment by carolynhu** [comment]
New issue, after @richardwxn fix, 1.5-release data is correct. But master data: `none_mtls_both` and`none_plaintext_both` is overlapping. 

@incfly http://perf.dashboard.istio.io/benchmarks/latency_vs_connection/
